L1Trigger/L1TTrackMatch/plugins/DisplacedVertexProducer.cc

```diff
@@ -0,0 +1,382 @@
+        bool isHard_j = false;
+        if (!tp_i->genParticles().empty() && !tp_j->genParticles().empty()) {
+          isHard_i = tp_i->genParticles()[0]->isHardProcess() || tp_i->genParticles()[0]->fromHardProcessFinalState();
+          isHard_j = tp_j->genParticles()[0]->isHardProcess() || tp_j->genParticles()[0]->fromHardProcessFinalState();
+        }
+
+        if (tp_i->eventId().event() == 0 && tp_j->eventId().event() == 0 && fabs(tp_i->vx() - tp_j->vx()) < 0.0001 &&
+            fabs(tp_i->vy() - tp_j->vy()) < 0.0001 && fabs(tp_i->vz() - tp_j->vz()) < 0.0001 && isHard_i && isHard_j &&
+            ((tp_i->charge() + tp_j->charge()) == 0)) {
+          isReal = true;
+        }
+      }
+
+      int inTraj =
+          calcVertex(selectedTracksWithTruth[i].first, selectedTracksWithTruth[j].first, x_dv_trk, y_dv_trk, z_dv_trk);
+      Vertex_Parameters vertex = Vertex_Parameters(
+          x_dv_trk, y_dv_trk, z_dv_trk, selectedTracksWithTruth[i].first, selectedTracksWithTruth[j].first);
+
+      if (vertex.R_T > RTMax_)
+        continue;
+      if (vertex.R_T < RTMin_)
+        continue;
+
+      l1t::DisplacedTrackVertex outputVertex = l1t::DisplacedTrackVertex(selectedTracksWithTruth[i].first.index,
+                                                                         selectedTracksWithTruth[j].first.index,
+                                                                         inTraj,
+                                                                         vertex.d_T,
+                                                                         vertex.R_T,
+                                                                         vertex.cos_T,
+                                                                         vertex.delta_z,
+                                                                         vertex.x_dv,
+                                                                         vertex.y_dv,
+                                                                         vertex.z_dv,
+                                                                         vertex.openingAngle,
+                                                                         vertex.p_mag,
+                                                                         isReal);
+
+      if (runEmulation_) {
+        std::vector<ap_fixed<13, 8, AP_RND_CONV, AP_SAT>> Transformed_features = {
+            selectedTracksWithTruth[i].first.pt * 0.25,
+            selectedTracksWithTruth[j].first.pt * 0.25,
+            selectedTracksWithTruth[i].first.eta,
+            selectedTracksWithTruth[j].first.eta,
+            selectedTracksWithTruth[i].first.phi,
+            selectedTracksWithTruth[j].first.phi,
+            selectedTracksWithTruth[i].first.d0,
+            selectedTracksWithTruth[j].first.d0,
+            selectedTracksWithTruth[i].first.z0,
+            selectedTracksWithTruth[j].first.z0,
+            selectedTracksWithTruth[i].first.chi2rz + 0.07,
```

Comment:
This conifer [PR](https://github.com/cms-sw/cmsdist/pull/9695) has now been merged so these additions of 0.07 shouldn't be necessary, can you check that the BDT is outputting as expected without these 0.07 additions and the new conifer external loaded

Reply:
Removing the shifts negatively impacts the scores for a small sample of signal events. The PR was merged to CMSSW 15_1_X and I have rebased to 15_0_X. So I guess I should rebase again so I get this change?

Reply:
I think in order to avoid the situation where your PR gets merged and then immediately broken by the 15_1 release you should wait until there is a 15_1 release and rebase to that

Reply:
Do you know when we can expect a 15_1 release? I don't believe my PR would be broken by this change. The shifting becomes pointless but the output should be basically the same